### PR TITLE
Tickets 119 and 122 [graph and pulldown fixes]

### PR DESCRIPTION
**Pull request by merlouduron**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Feature addition
    - [ ] Bug fix
    - [ ] Refactoring
    - [ ] Documentation update
    - [ ] Other (please specify below)

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: client/src/pages/ResetPassword/ResetPassword.tsx [hunk 1]
```diff
@@ -22,8 +22,8 @@ const ResetPassword = () => {
         }
 
         try {
-            // const response = await axios.put(`http://127.0.0.1:8000/api/reset-password/${uid}/${token}/`, {
-            const response = await axios.put(`http://54.178.202.58:8000/api/reset-password/${uid}/${token}/`, {
+            const response = await axios.put(`http://127.0.0.1:8000/api/reset-password/${uid}/${token}/`, {
+            // const response = await axios.put(`http://54.178.202.58:8000/api/reset-password/${uid}/${token}/`, {
                 password
             });
             alert("Password has been resetted");
```
Comment (edward-stalley-work) [inline client/src/pages/ResetPassword/ResetPassword.tsx:29]: Please change to:
→ ”Password has been reset."

Comment (edward-stalley-work) [reply client/src/pages/ResetPassword/ResetPassword.tsx:29]: Is it possible to also use the modals that Joshua created and that are being used for all other messages?

Comment (edward-stalley-work) [reply client/src/pages/ResetPassword/ResetPassword.tsx:29]: Once the Password has been reset we should automatically move to login screen after a second or two.
As there is no link/tab to move to login from this screen.

File: client/src/reducers/card/cardSlice.ts [hunk 2]
```diff
@@ -50,22 +39,10 @@ const cardSlice = createSlice({
...
-        )
-        state.totalGrossProfitMargin = getOtherPlanningSum(
-          action?.payload.map((card) => card.other_planning),
-          'gross_profit_margin',
-        )
-        state.totalOperatingProfitMargin = getOtherPlanningSum(
-          action?.payload.map((card) => card.other_planning),
-          'operating_profit_margin',
-        )
+        state.totalGrossProfit = getSum(action?.payload?.map((card) => Number(card.ordinary_profit)))
```
Comment (edward-stalley-work) [inline client/src/reducers/card/cardSlice.ts:42]: i think we need to change ordinary_profit to ordinary_income.

File: client/src/reducers/card/cardSlice.ts [hunk 3]
```diff
@@ -50,22 +39,10 @@ const cardSlice = createSlice({
...
+        state.totalNetProfitPeriod = getSum(action?.payload?.map((card) => Number(card.non_operating_profit)))
+        state.totalGrossProfitMargin = getSum(action?.payload?.map((card) => Number(card.ordinary_profit_margin)))
+        state.totalOperatingProfitMargin = getSum(action?.payload?.map((card) => Number(card.non_operating_expense)))
       })
       .addCase(fetchAllCards.pending, (state) => {
         state.isLoading = true
```
Comment (edward-stalley-work) [inline client/src/reducers/card/cardSlice.ts:48]: I am not 100% sure myself so I would like to discuss these values with you. We need them for the planning page as well so I would just like to make sure we are displaying correct values.

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
I was not sure if you had 100% finished but I took a quick look anyway.
please see comments.

File: client/src/pages/ResetPassword/ResetPassword.tsx (hunk 1 above)
Comment (merlouduron) [reply client/src/pages/ResetPassword/ResetPassword.tsx]: The message for resetting the password has been updated.
It now uses modals like other pages.
It also navigates to the login page after successfully resetting the password.

File: client/src/reducers/card/cardSlice.ts (hunk 2 above)
Comment (merlouduron) [reply client/src/reducers/card/cardSlice.ts:42]: I will update this after the completion of the planning page.

File: client/src/reducers/card/cardSlice.ts (hunk 3 above)
Comment (merlouduron) [reply client/src/reducers/card/cardSlice.ts:48]: I will update this after the completion of the planning page.

File: client/src/pages/ResetPassword/ResetPassword.tsx [hunk 4]
```diff
@@ -1,38 +1,65 @@
-import React, { useState } from "react";
```
Comment (edward-stalley-work) [inline client/src/pages/ResetPassword/ResetPassword.tsx:1]: Currently I am no receiving an email to reset password.
It did work previously. Can you doublecheck it is working your end? 

I am using my email for the user in auth_user table.
It says it sends - however my inbox does not have a message.

Screenshot 2024-10-29 at 10 33 47

Previously I successfully got this email.

Screenshot 2024-10-29 at 10 34 40

File: client/src/reducers/graph/graphSlice.ts [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-  totalSalesByDate: Record<string, number>;
-  totalOperatingProfitByDate: Record<string, number>;
-  totalGrossProfitByDate: Record<string, number>;
-  totalNetProfitPeriodByDate: Record<string, number>;
-  totalGrossProfitMarginByDate: Record<string, number>;
-  totalOperatingProfitMarginByDate: Record<string, number>;
-  datePlanning: string[];
+  isLoading: boolean
+  totalSalesByDate: Record<string, number>
+  totalOperatingProfitByDate: Record<string, number>
```
Comment (edward-stalley-work) [inline client/src/reducers/graph/graphSlice.ts:8]: For now to keep it in line with name in other parts of code and　
→totalOperatingIncomeByDate

Please adjust  any related code in this file. Also in Dashboard.tsx as well

File: client/src/reducers/graph/graphSlice.ts [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
+  totalGrossProfitByDate: Record<string, number>
+  totalNetProfitPeriodByDate: Record<string, number>
```
Comment (edward-stalley-work) [inline client/src/reducers/graph/graphSlice.ts:10]: totalNetProfitPeriodByDate → totalCumulativeOrdinaryIncome

Please adjust  any related code in this file. Also in Dashboard.tsx as well

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Thank you Merlou please see comments

File: backend/api/models.py [hunk 7]
```diff
@@ -160,7 +160,7 @@ class Projects(models.Model):
     non_operating_expense = models.DecimalField(
         max_digits=12, decimal_places=2, default=0.0
     )
-    ordinary_profit = models.DecimalField(max_digits=12, decimal_places=2, default=0.0)
+    ordinary_income = models.DecimalField(max_digits=12, decimal_places=2, default=0.0)
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py]: Fixed.

File: client/src/pages/ResetPassword/ResetPassword.tsx (hunk 4 above)
Comment (merlouduron) [reply client/src/pages/ResetPassword/ResetPassword.tsx:1]: For this one, if it doesn't exist on the inbox, please do check on spam folder. If still doesn't exist, the problem is on the software used. I've tested my email and it also rejected. Will probably use another software.

File: client/src/reducers/graph/graphSlice.ts (hunk 5 above)
Comment (merlouduron) [reply client/src/reducers/graph/graphSlice.ts]: Fixed.

File: client/src/reducers/graph/graphSlice.ts (hunk 6 above)
Comment (merlouduron) [reply client/src/reducers/graph/graphSlice.ts]: Fixed.

File: client/src/pages/ResetPassword/ResetPassword.tsx (hunk 4 above)
Comment (edward-stalley-work) [reply client/src/pages/ResetPassword/ResetPassword.tsx:1]: still does no not receive an email (not in inbox or spam).

It used to work - did the mail software you are using change?

**Review by edward-stalley-work — COMMENTED** [review]
thank you merlou - please see new comments.

The email still does not work unfortunately.

File: client/src/pages/ResetPassword/ResetPassword.tsx (hunk 1 above)
Comment (edward-stalley-work) [reply client/src/pages/ResetPassword/ResetPassword.tsx]: Due to issues with mail software this no longer works and will be tackled in another ticket once live server is up and running.

File: client/src/pages/ResetPassword/ResetPassword.tsx (hunk 4 above)
Comment (edward-stalley-work) [reply client/src/pages/ResetPassword/ResetPassword.tsx:1]: Due to issues with mail software this no longer works and will be tackled in another ticket once live server is up and running.

**Review by edward-stalley-work — APPROVED** [review]

**Pull request merged**